chainerx_cc/chainerx/cuda/cuda_device/rnn.cu

```diff
@@ -0,0 +1,502 @@
+#include "chainerx/cuda/cuda_device.h"
+
+#include <algorithm>
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include <absl/types/optional.h>
+#include <cuda.h>
+#include <cudnn.h>
+
+#include "chainerx/array.h"
+#include "chainerx/backend_util.h"
+#include "chainerx/constant.h"
+#include "chainerx/cuda/cuda.h"
+#include "chainerx/cuda/cuda_backend.h"
+#include "chainerx/cuda/cuda_runtime.h"
+#include "chainerx/cuda/cuda_set_device_scope.h"
+#include "chainerx/cuda/cudnn.h"
+#include "chainerx/cuda/kernel_regist.h"
+#include "chainerx/device.h"
+#include "chainerx/dtype.h"
+#include "chainerx/error.h"
+#include "chainerx/kernels/connection.h"
+#include "chainerx/kernels/creation.h"
+#include "chainerx/kernels/misc.h"
+#include "chainerx/kernels/rnn.h"
+#include "chainerx/macro.h"
+#include "chainerx/native/kernel_regist.h"
+#include "chainerx/routines/connection.h"
+#include "chainerx/routines/creation.h"
+#include "chainerx/routines/manipulation.h"
+#include "chainerx/shape.h"
+#include "chainerx/stack_vector.h"
+
+namespace chainerx {
+namespace cuda {
+namespace {
+
+__global__ void InitGPUDataKer(float* data, int64_t numElements, float* value) {
```

Comment:
Use `snake_case` for variable names: `numElements` -> `num_elements`, `gridDim` -> `grid_dim`, `wDesc_` -> `w_desc_` etc.